### ISSUE TITLE
Unsupported attribute / empty tuple when removing count

### ISSUE THREAD

**Issue by TomTucka** [hashicorp/terraform#25569]
### Terraform Version

```
λ terraform version
Terraform v0.12.28
+ provider.archive v1.3.0
+ provider.aws v2.70.0
```

### Terraform Configuration Files

```terraform
resource "aws_dynamodb_table" "file_service_requests" {
  count       = 1
  name         = "file-service-requests-${terraform.workspace}"
  hash_key     = "Ref"
  billing_mode = "PAY_PER_REQUEST"

  attribute {
    name = "Ref"
    type = "S"
  }

  ttl {
    attribute_name = "Ttl"
    enabled        = true
  }
}

output "dynamodb_table_name" {
  value       = aws_dynamodb_table.file_service_requests[0].name
  description = "Name of the dynamodb for inbound requests"
}
```

After removing count: 
```terraform
resource "aws_dynamodb_table" "file_service_requests" {
  name         = "file-service-requests-${terraform.workspace}"
  hash_key     = "Ref"
  billing_mode = "PAY_PER_REQUEST"

  attribute {
    name = "Ref"
    type = "S"
  }

  ttl {
    attribute_name = "Ttl"
    enabled        = true
  }
}

output "dynamodb_table_name" {
  value       = aws_dynamodb_table.file_service_requests.name
  description = "Name of the dynamodb for inbound requests"
}
```

### Expected Behavior

The plan should not show any changes and should not fail

### Actual Behavior

Plan fails:

```
Error: Unsupported attribute

  on modules/file_service/outputs.tf line 7, in output "dynamodb_table_name":
   7:   value       = aws_dynamodb_table.file_service_requests.name
    |----------------
    | aws_dynamodb_table.file_service_requests is empty tuple

This value does not have any attributes.

```
### Steps to Reproduce

- [ ] terraform init
- [ ] terraform apply
- [ ] remove count
- [ ] terraform plan

### References

- Seems related to #23676

**Comment by jbardin** [comment]
I've confirmed this is fixed in the latest 0.13 release.
Thanks for reporting it @TomTucka!

**Comment by pselle** [comment]
0.13 has been released: https://www.terraform.io/downloads.html
